apis/nucleus/src/components/listbox/listbox-highlight.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @typedef {object} Range
+ * @property {number} qCharPos The (absolute) index where the highlighted range starts.
+ * @property {number} qCharCount The length of the sub-string (starting from qChartPos) that should be highlighted.
+ */
+
+/**
+ * @typedef {Array<{segment: string, highlighted: boolean}>} Segment
```

Comment:
Defining objects inline will cause problems when generating typescript definitions. It looks like the typedef can be changed to:

```
@typedef {object} Segment
```

And when using it around line 18:

```
@returns {Segment[]} The segments
```

Reply:
I will change.